Flakyness in ert/tests/ert_tests/dark_storage/test_graphql_queries.py

Observed flakyness in https://github.com/equinor/ert/runs/5537967474?check_suite_focus=true

Python 3.8, on ubuntu-latest.

While running `test_graphql_queries.py` (seen from the last thread stacktrace) the output is

```
Abort called from: buffer_fseek (/github/workspaAbort called from: buffer_fseek (/github/workspace/lib/util/buffer.cpp:244)

Error message: buffer_fseek: tried to seek to position:16 - outside of bounds: [0,0) 


--------------------------------------------------------------------------------
Stack trace (most recent call last) in thread 7026:
#10   Object "", at 0xffffffffffffffff, in 
#9    Object "/lib/x86_64-linux-gnu/libc.so.6", at 0x7f21284ec292, in clone
#8    Object "/lib/x86_64-linux-gnu/libpthread.so.0", at 0x7f21283b0608, in start_thread
#7    Object "/lib/x86_64-linux-gnu/libstdc++.so.6", at 0x7f2124269de3, in std::error_code::default_error_condition() const
#6    Object "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/res/_lib.cpython-38-x86_64-linux-gnu.so", at 0x7f2124005939, in std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::thread::_Invoker<std::tuple<void (*)(bfs_struct*), bfs_struct*> >, void> >::_M_invoke(std::_Any_data const&)
#5    Object "/lib/x86_64-linux-gnu/libpthread.so.0", at 0x7f21283b947e, in __pthread_once_slow
#4    Object "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/res/_lib.cpython-38-x86_64-linux-gnu.so", at 0x7f2123fe2a3a, in std::__future_base::_State_baseV2::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*)
#3    Object "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/res/_lib.cpython-38-x86_64-linux-gnu.so", at 0x7f21240082ea, in ert::block_fs_driver::mount()
#2    Object "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/res/_lib.cpython-38-x86_64-linux-gnu.so", at 0x7f2123f8d25b, in block_fs_mount(char const*, int, float, int, bool, bool)
#1    Object "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/ecl/.libs/libecl.so", at 0x7f21243f6da4, in buffer_fseek
#0    Object "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/ecl/.libs/libecl.so", at 0x7f21243e5cae, in util_abort__
--------------------------------------------------------------------------------

Fatal Python error: Aborted

Thread 0x00007f2094ff8700 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/cwrap/prototype.py", line 218 in __call__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/res/enkf/enkf_main.py", line [275](https://github.com/equinor/ert/runs/5537967474?check_suite_focus=true#step:7:275) in __init__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/res/enkf/enkf_main.py", line 72 in __init__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/ert_shared/dark_storage/enkf.py", line 27 in init_facade
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/ert_shared/dark_storage/enkf.py", line 51 in get_res
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/ert_shared/dark_storage/graphql/experiments.py", line 50 in resolve_ensembles
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executors/sync.py", line 16 in execute
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 452 in resolve_or_error
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 365 in resolve_field
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 244 in execute_fields
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 722 in complete_object_value
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 571 in complete_value
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 480 in complete_value_catching_error
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 600 in complete_list_value
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 556 in complete_value
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 480 in complete_value_catching_error
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 367 in resolve_field
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 244 in execute_fields
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 182 in execute_operation
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 127 in promise_executor
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/promise.py", line 87 in try_catch
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/promise.py", line 347 in _settle_promise_from_handler
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/promise.py", line 323 in _settle_promise
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/async_.py", line 80 in drain_queue
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/async_.py", line 122 in drain_queues
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/schedulers/immediate.py", line 12 in call
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/async_.py", line 131 in queue_tick
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/async_.py", line 34 in _async_invoke
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/async_.py", line 50 in invoke
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/promise.py", line 588 in _then
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/promise/promise.py", line 630 in then
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/execution/executor.py", line 145 in execute
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/backend/core.py", line 32 in execute_and_validate
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/graphql.py", line 64 in execute_graphql
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphql/graphql.py", line 44 in graphql
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/graphene/types/schema.py", line 144 in execute
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 754 in run
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f2093ff7700 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/selectors.py", line 468 in select
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/asyncio/base_events.py", line 1823 in _run_once
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/asyncio/base_events.py", line 570 in run_forever
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/asyncio/base_events.py", line 603 in run_until_complete
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/asyncio/runners.py", line 44 in run
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 233 in run
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/anyio/_core/_eventloop.py", line 56 in run
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/concurrent/futures/thread.py", line 57 in run
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/concurrent/futures/thread.py", line 80 in _worker
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/threading.py", line 870 in run
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f20f9c0a700 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/cwrap/prototype.py", line 218 in __call__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/res/enkf/enkf_fs.py", line 130 in umount
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/res/enkf/enkf_fs.py", line 137 in free
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/cwrap/basecclass.py", line 137 in __del__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/_weakrefset.py", line 59 in __iter__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/_weakrefset.py", line 119 in update
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/_weakrefset.py", line 50 in __init__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/_weakrefset.py", line 92 in copy
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/tqdm/_monitor.py", line 49 in get_instances
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/tqdm/_monitor.py", line 69 in run
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f21[282](https://github.com/equinor/ert/runs/5537967474?check_suite_focus=true#step:7:282)48740 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/threading.py", line [302](https://github.com/equinor/ert/runs/5537967474?check_suite_focus=true#step:7:302) in wait
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/concurrent/futures/_base.py", line 439 in result
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/anyio/from_thread.py", line 240 in call
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/starlette/testclient.py", line 263 in send
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/requests/sessions.py", line 645 in send
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/requests/sessions.py", line 529 in request
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/starlette/testclient.py", line 468 in request
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/requests/sessions.py", line 577 in post
  File "/home/runner/work/ert/ert/tests/ert_tests/dark_storage/test_graphql_queries.py", line 87 in test_query_ensemble_parameters
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/python.py", line 192 in pytest_pyfunc_call
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/python.py", line 1761 in runtest
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/runner.py", line 166 in pytest_runtest_call
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/runner.py", line 259 in <lambda>
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/runner.py", line 338 in from_call
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/runner.py", line 258 in call_runtest_hook
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/flaky/flaky_pytest_plugin.py", line 138 in call_and_report
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/runner.py", line 130 in runtestprotocol
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/runner.py", line 111 in pytest_runtest_protocol
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/flaky/flaky_pytest_plugin.py", line 94 in pytest_runtest_protocol
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/main.py", line 347 in pytest_runtestloop
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/main.py", line 322 in _main
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/main.py", line 268 in wrap_session
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/main.py", line [315](https://github.com/equinor/ert/runs/5537967474?check_suite_focus=true#step:7:315) in pytest_cmdline_main
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/config/__init__.py", line 164 in main
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/config/__init__.py", line 187 in console_main
  File "/opt/hostedtoolcache/Python/3.8.12/x64/bin/pytest", line 8 in <module>
/home/runner/work/_temp/6ec980b5-f56a-4c29-b37a-62db81389dab.sh: line 4:  [319](https://github.com/equinor/ert/runs/5537967474?check_suite_focus=true#step:7:319)3 Aborted                 (core dumped) pytest ert_tests -sv --durations=0 -m "not integration_test"
Error: Process completed with exit code 134.
```

--- a/tests/ert_tests/dark_storage/conftest.py
+++ b/tests/ert_tests/dark_storage/conftest.py
@@ -5,6 +5,7 @@ import pytest
 import os
 import shutil
 import py
+import gc
 
 from starlette.testclient import TestClient
 
@@ -76,6 +77,7 @@ def reset_enkf():
     enkf._config = None
     enkf._ert = None
     enkf._libres_facade = None
+    gc.collect()
 
 
 def new_storage_client(monkeypatch, ert_storage_client):

